staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -412,6 +412,15 @@ func (s *store) GetToList(ctx context.Context, key string, resourceVersion strin
 	return s.versioner.UpdateList(listObj, uint64(getResp.Header.Revision), "")
 }
 
+func (s *store) Count(key string) (int64, error) {
+	key = path.Join(s.pathPrefix, key)
+	getResp, err := s.client.KV.Get(context.Background(), key, clientv3.WithRange(clientv3.GetPrefixRangeEnd(key)), clientv3.WithCountOnly())
```

Comment:
we should not issue large range to etcd. just to count the keys, we should use keys only option with limit and maybe also with pagination.

Reply:
is `WithCountOnly` not more efficient than `WithKeysOnly`?

Reply:
oh. i did not see count only is already there. but still if we know the count is going to be huge, probably need pagination as well. etcd does not maintain a index for count(*), but scan everything internally.

Reply:
doing a micro-benchmark could be helpful. if count 1M keys take <1s, it might be ok.

Reply:
If we would have 1M keys, we would have other problems than this metric...

Reply:
counting 1M keys just to make the test result more reliable and safer. you can divide that by 10 to get 100k for an upper bound, etc.. periodically blocking etcd for more than 50ms is not good in general. i have not really looked into this pr. so just some random suggestions and thoughts on this.

Reply:
I can run benchmarks after the freeze, as it's not enough time now. It shouldn't be an issue, as it's a 1-word change to disable this functionality.

As for pagination I'm not sure I follow. I believe that etcd-side 'WithLimit' restricts response size (i.e. number of items returned), which shouldn't apply to "count only" requests, as they always return single value. Or is the 'countOnly' implemented on the client side, thus "limit" will effectively cap the returned value to, well, limit?

Or maybe you meant doing some kind of manual pagination by dividing the range into multiple subranges, querying them separately and combining the result?